ocs_ci/helpers/dr_helpers.py

```diff
@@ -285,27 +346,101 @@ def wait_for_vr_creation(count, namespace, timeout=300):
+    sample = TimeoutSampler(
+        timeout=timeout,
+        sleep=3,
+        func=check_vr_state,
+        state="primary",
+        namespace=namespace,
+    )
+    assert sample.wait_for_func_status(
+        result=True
+    ), "One or more VR haven't reached expected state primary within the time limit."
+
+    logger.info("Waiting for VRG to reach primary state")
+    sample = TimeoutSampler(
+        timeout=timeout,
+        sleep=3,
+        func=check_vrg_state,
+        state="primary",
+        namespace=namespace,
+    )
+    assert sample.wait_for_func_status(
+        result=True
+    ), "VRG hasn't reached expected state primary within the time limit."
 
 
-def wait_for_vr_deletion(namespace, timeout=300):
+def wait_for_replication_resources_deletion(namespace, timeout, check_state=True):
     """
-    Wait for all VR resources to be deleted in the given namespace
+    Wait for replication resources to be deleted
 
     Args:
         namespace (str): the namespace of the VR resources
-        timeout (int): time in seconds to wait for VR resources to be created
-            or reach expected state
+        timeout (int): time in seconds to wait for VR resources to reach expected
+            state or deleted
+        check_state_before_deletion (bool): True for checking vr state before deletion, False otherwise
 
     """
-    wait_for_vr_state("secondary", namespace, timeout)
-    wait_for_vr_count(0, namespace, timeout)
+    if check_state:
+        logger.info("Waiting for all VRs to reach secondary state")
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=3,
+            func=check_vr_state,
+            state="secondary",
+            namespace=namespace,
+        )
+        assert sample.wait_for_func_status(
+            result=True
+        ), "One or more VR haven't reached expected state secondary within the time limit."
+
+        logger.info("Waiting for VRG to reach secondary state")
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=3,
+            func=check_vrg_state,
+            state="secondary",
+            namespace=namespace,
+        )
+        assert sample.wait_for_func_status(
+            result=True
+        ), "VRG hasn't reached expected state secondary within the time limit."
+
+    logger.info("Waiting for VRG to be deleted")
+    sample = TimeoutSampler(
+        timeout=timeout, sleep=3, func=check_vrg_existence, namespace=namespace
+    )
+    assert sample.wait_for_func_status(result=False), "VRG resource is not deleted"
+
+    logger.info("Waiting for all VRs to be deleted")
+    sample = TimeoutSampler(
+        timeout=timeout,
+        sleep=3,
+        func=get_vr_count,
+        namespace=namespace,
+    )
+    sample.wait_for_func_value(0)
 
 
-def wait_for_workload_resource_creation(pvc_count, pod_count, namespace, timeout=120):
+def wait_for_all_resources_creation(pvc_count, pod_count, namespace, timeout=900):
```

Comment:
@keesturam @am-agrawa jfyi, currently the timeout for resource creation is set to 900 seconds